shared/js/project.js

```diff
@@ -309,13 +357,20 @@ var projects = module.exports = {
    */
   load: function () {
     var deferred;
-    if (appOptions.level.isProjectLevel) {
+    if (projects.isProjectLevel()) {
       if (redirectFromHashUrl() || redirectEditView()) {
         return;
       }
       var pathInfo = parsePath();
 
-      if (pathInfo.channelId) {
+      // External levels which have a channel id should fetch the data in that
+      // channel from the server. In other cases, appOptions.channel is assumed to
+      // contain no data and will only be used later when data is saved back
+      // to the server.
+      var channelId = pathInfo.channelId ||
+          (appOptions.isExternalProjectLevel && appOptions.channel);
```

Comment:
Good news @joshlory , your intuition was right -- this usage of isExternalProjectLevel is unnecessary (this code never even gets executed for pixelation). I've now reverted this part of the change.